scripts/check-adoption-census-schema-contract.py

```
@@ -0,0 +1,191 @@
+    current_path: Path,
+    documents: dict[Path, dict[str, Any]],
+) -> tuple[dict[str, Any], Path]:
+    filename, separator, fragment = reference.partition("#")
+    target_path = current_path if not filename else (current_path.parent / filename).resolve()
+    target = documents.get(target_path)
+    if target is None:
+        raise AssertionError(f"missing schema reference {reference} from {current_path}")
+    return (pointer(target, f"#{fragment}") if separator else target), target_path
+
+
+def matches_type(value: Any, expected: str) -> bool:
```

Ruff (0.16.2)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `value`

(ANN401)